src/oai/params.ts

```diff
@@ -0,0 +1,67 @@
+import { FunctionPayload } from "./fns"
+
+export function OAIBuildFunctionParams(definition: FunctionPayload, params) {
+  return {
+    ...params,
+    function_call: {
+      name: definition.name
+    },
+    functions: [
+      ...(params?.functions ?? []),
+      {
+        name: definition.name,
+        description: definition.description,
+        parameters: {
+          type: "object",
+          properties: definition.parameters,
+          required: definition.required
+        }
+      }
+    ]
+  }
+}
+
+export function OAIBuildToolFunctionParams(definition: FunctionPayload, params) {
+  return {
+    ...params,
+    tool_choice: {
+      type: "function",
+      function: { name: definition.name }
+    },
+    tools: [
+      ...(params?.tools ?? []),
+      {
+        type: "function",
+        function: {
+          name: definition.name,
+          description: definition.description,
+          parameters: {
+            type: "object",
+            properties: definition.parameters,
+            required: definition.required
+          }
+        }
+      }
+    ]
+  }
+}
+
+export function OAIBuildMessageBasedParams(definition: FunctionPayload, params) {
```

Comment:
MD_JSON
JSON
JSON_SCHEMA are different modes 

and the parameters are different. 

```python
if mode == Mode.JSON:
   new_kwargs["response_format"] = {"type": "json_object"}

elif mode == Mode.JSON_SCHEMA:
    new_kwargs["response_format"] = {
        "type": "json_object",
        "schema": response_model.model_json_schema(),
    }

elif mode == Mode.MD_JSON:
    new_kwargs["messages"].append(
        {
            "role": "assistant",
            "content": "```json",
        },
    )
    new_kwargs["stop"] = "```"
    ```